lib/rspec/core.rb

```diff
@@ -56,6 +56,7 @@ class << self
   # they use the runner multiple times within the same process. Users must deal
   # themselves with re-configuration of RSpec before run.
   def self.reset
+    @world.example_groups.reduce(&:remove_all_constants)
```

Comment:
Actually, on further reflection, I realized that we don't want to apply a side effect to each group in `@world.example_groups` after all.  The suggestion about favoring `each` over `reduce` when only applying a side effect to every element is a good general tip, though.